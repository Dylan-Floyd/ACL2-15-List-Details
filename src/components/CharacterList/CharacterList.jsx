import React from 'react'
import { Link } from 'react-router-dom'
import Character from '../Character/Character.jsx'
import styles from './CharacterList.module.css'

export default function CharacterList({ characters }) {

  return (
    <div className={styles.charList}>
      {
        characters.map(char => (
          <Link to={`/characters/${char.id}`} className={styles.clickable} key={char.id} >
            <Character characterData={char} />
          </Link>
        ))
      }
    </div>
  )
}
