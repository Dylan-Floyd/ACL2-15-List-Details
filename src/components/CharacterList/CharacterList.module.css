.charList {
  display: flex;
  flex-flow: row wrap;
  gap: 20px;
}

.clickable {
  text-decoration: none;
}