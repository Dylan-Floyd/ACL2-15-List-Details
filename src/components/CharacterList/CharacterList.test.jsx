import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import CharacterList from './CharacterList.jsx'

it('should render a list of characters', () => {
  const characters = [
    {
      id: 1,
      name: 'Rick Sanchez',
      status: 'Alive',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
    },
    {
      id: 2,
      name: 'Rick Sanchez2',
      status: 'Alive',
      image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg'
    }
  ]

  const { container } = render(
    <MemoryRouter>
      <CharacterList characters={characters} />
    </MemoryRouter>
  )
  expect(container).toMatchSnapshot()
})