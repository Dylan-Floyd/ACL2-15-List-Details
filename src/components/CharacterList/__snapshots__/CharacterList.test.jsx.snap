// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of characters 1`] = `
<div>
  <div
    class="charList"
  >
    <a
      class="clickable"
      href="/characters/1"
    >
      <div
        class="charCard"
      >
        <h3
          class="name"
        >
          Rick Sanchez
        </h3>
        <div
          class="origin"
        >
          Origin: 
        </div>
        <div>
          Status: 
          Alive
        </div>
        <img
          alt="Rick Sanchez"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
      </div>
    </a>
    <a
      class="clickable"
      href="/characters/2"
    >
      <div
        class="charCard"
      >
        <h3
          class="name"
        >
          Rick Sanchez2
        </h3>
        <div
          class="origin"
        >
          Origin: 
        </div>
        <div>
          Status: 
          Alive
        </div>
        <img
          alt="Rick Sanchez2"
          src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
        />
      </div>
    </a>
  </div>
</div>
`;
